plugins/woocommerce/tests/php/src/Internal/DataStores/StockNotifications/StockNotificationsDataStoreTests.php

```diff
@@ -0,0 +1,535 @@
+	}
+
+	/**
+	 * Test deleting a notification.
+	 */
+	public function test_delete_notification() {
+		// Create a notification.
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->save();
+
+		$notification_id = $notification->get_id();
+		$this->assertGreaterThan( 0, $notification_id );
+
+		// Delete the notification.
+		$notification->delete();
+
+		// Verify the notification was deleted.
+		$this->assertEquals( 0, $notification->get_id() );
+
+		// Try to read the deleted notification.
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'Stock notification not found' );
+		new Notification( $notification_id );
+	}
+
+	/**
+	 * Test reading a non-existent notification.
+	 */
+	public function test_read_nonexistent_notification() {
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'Stock notification not found' );
+		new Notification( 999999 );
+	}
+
+	/**
+	 * Test adding a meta to a notification.
+	 */
+	public function test_add_meta_to_notification() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+
+		// Refresh the notification.
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+	}
+
+	/**
+	 * Test updating a meta for a notification.
+	 */
+	public function test_update_meta_for_notification() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->save();
+
+		// Refetch the notification.
+		$notification = new Notification( $notification->get_id() );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+		$notification->update_meta_data( 'test_meta', 'updated_value' );
+		$notification->save();
+
+		// Refetch the notification.
+		$notification = new Notification( $notification->get_id() );
+
+		$this->assertEquals( 'updated_value', $notification->get_meta( 'test_meta' ) );
+	}
+
+	/**
+	 * Test deleting a meta for a notification.
+	 */
+	public function test_delete_meta_for_notification() {
+		$notification = new Notification();
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->add_meta_data( 'test_meta', 'test_value' );
+		$notification->save();
+
+		$notification = new Notification( $notification->get_id() );
+		$this->assertEquals( 'test_value', $notification->get_meta( 'test_meta' ) );
+		$notification->delete_meta_data( 'test_meta' );
+		$notification->save();
```

Comment:
Might be good to check non-existence also before re-reading it below.

Reply:
Good call, added in https://github.com/woocommerce/woocommerce/pull/57062/commits/156b4580eade80e13961c22fcd0aa3e3729d34e7